src/crimson/admin/admin_socket.cc

```
@@ -0,0 +1,646 @@
+                                   ceph::bufferlist& out) const
+{
+  // did we receive any arguments apart from the command word(s)?
+  if (parsed.m_cmd_seq_len == parsed.m_cmd.length())
+    return true;
+
+  logger().info("{}: validating {} against:{}", __func__, command_text,
+                parsed.m_api->cmddesc);
+
+  stringstream os;  // for possible validation error messages
+  try {
+    // validate_cmd throws on some syntax errors
+    if (validate_cmd(nullptr, parsed.m_api->cmddesc, parsed.m_parameters, os)) {
+      return true;
+    }
+  } catch (std::exception& e) {
+    logger().error("{}: validation failure ({} : {}) {}", __func__,
+                   command_text, parsed.m_cmd, e.what());
+  }
+
+  os << "error: command validation failure ";
+  logger().error("{}: validation failure (incoming:{}) {}", __func__,
+                 command_text, os.str());
+  out.append(os);
+  return false;
+}
+
+seastar::future<> AdminSocket::finalize_response(
+  seastar::output_stream<char>& out, ceph::bufferlist&& msgs)
+{
+  string outbuf_cont = msgs.to_str();
+  if (outbuf_cont.empty()) {
+    outbuf_cont = " {} ";
+  }
+  uint32_t response_length = htonl(outbuf_cont.length());
+  logger().info("asok response length: {}", outbuf_cont.length());
+
+  return out.write((char*)&response_length, sizeof(uint32_t))
+    .then([&out, outbuf_cont] { return out.write(outbuf_cont.c_str()); });
+}
+
+seastar::future<> AdminSocket::execute_line(std::string cmdline,
+                                            seastar::output_stream<char>& out)
+{
+  return seastar::with_shared(
+    servers_tbl_rwlock, [this, cmdline, &out]() mutable {
+      ceph::bufferlist msgs;
+      maybe_parsed_t oparsed;
+
+      bool ok_to_execute{ false };
+      try {
+        oparsed = parse_cmd(cmdline, msgs);
+        if (oparsed.has_value())
+          ok_to_execute = validate_command(*oparsed, cmdline, msgs);
+      } catch (...) {
```

what exception is expected here? and why can we ignore the exception?

---

@tchaikov 
validate_command() has some paths that throw on failure (I would have prefered it to simply
return an error value in those cases). And a failure to set ok_to_execute, for whatever reason (i.e. either in parse_cmd() or in validate_cmd()) is handled.

---

could you point me to the line where the exception is thrown?

---

Noticed that I already handle validate_cmd() exceptions within validate_command(), so you are right.